gaia-fast-list.js

```diff
@@ -558,10 +569,25 @@ Internal.prototype = {
     this.model = model;
     this.sections = this.sectionize(model);
     if (!this.fastList) this.createList();
-    else this.fastList.reloadData();
+    else this.reloadData();
     this.setEmpty(false);
   },
 
+  /**
+   * Reloading the list will completely
+   * re-render the list.
+   *
+   * We must empty the image cache as data
+   * may have changed or item indexed
+   * may not longer map to new model.
+   *
+   * @private
+   */
+  reloadData() {
+    this.emptyImageCache();
```

Comment:
might be worth it to file follow up to add a formal way to "append" content (ie. new content bellow but the content that was there didn't move). We'll be able to do a bunch of optimizations, including not throwing away the image cache.